Source/SwiftLintCore/Extensions/Configuration+FileGraph.swift

```diff
@@ -89,18 +89,29 @@ public extension Configuration {
             )
 
             if !ignoreParentAndChildConfigs {
-                try processPossibleReference(
-                    ofType: .childConfig,
-                    from: vertex,
-                    remoteConfigTimeoutOverride: remoteConfigTimeoutOverride,
-                    remoteConfigTimeoutIfCachedOverride: remoteConfigTimeoutIfCachedOverride
-                )
-                try processPossibleReference(
-                    ofType: .parentConfig,
-                    from: vertex,
-                    remoteConfigTimeoutOverride: remoteConfigTimeoutOverride,
-                    remoteConfigTimeoutIfCachedOverride: remoteConfigTimeoutIfCachedOverride
-                )
+                do {
+                    try processPossibleReference(
+                        ofType: .childConfig,
+                        from: vertex,
+                        remoteConfigTimeoutOverride: remoteConfigTimeoutOverride,
+                        remoteConfigTimeoutIfCachedOverride: remoteConfigTimeoutIfCachedOverride
+                    )
+                    try processPossibleReference(
```

Comment:
I think you should have both in a separate try-catch. Otherwise, we don't check the parent configuration when there was a problem with reading a child configuration.

Reply:
Good catch. Split into two try/catch

Reply:
Please extract this to a function.